spec/unit/util/storage_spec.rb

```diff
@@ -220,16 +221,12 @@ def write_state_file(contents)
       stale_checked = Time.now - (Puppet[:statettl] + 1)
       Puppet::Util::Storage.cache(:yayness)[:checked] = recent_checked
       Puppet::Util::Storage.cache(:stale)[:checked] = stale_checked
-      expect(Puppet::Util::Storage.state).to eq(
-        {
-          :yayness => {
-            :checked => recent_checked
-          },
-          :stale => {
-            :checked => stale_checked
-          }
-        }
-      )
+
+      expect(Puppet::Util::Storage.state.keys).to contain_exactly(:yayness, :stale)
+      expect(Puppet::Util::Storage.state[:yayness].keys).to contain_exactly(:checked)
+      expect(Puppet::Util::Storage.state[:stale].keys).to contain_exactly(:checked)
+      expect(Puppet::Util::Storage.state[:yayness][:checked]).to be_within(@A_SMALL_AMOUNT_OF_TIME).of(recent_checked)
+      expect(Puppet::Util::Storage.state[:stale][:checked]).to be_within(@A_SMALL_AMOUNT_OF_TIME).of(stale_checked)
```

Comment:
Wondering if you could use composables matchers to match the hash while allowing a range of values, eg

```ruby
expect(..).to match(
  {
    :yayness => {
      :checked => a_value_within(0.001).of(recent_checked)
    },
    ...
  }
```

Reply:
I'll try it.

Reply:
Worked great!